cmd/swarm/upload.go

```diff
@@ -119,7 +118,7 @@ func upload(ctx *cli.Context) {
 			}
 			defer f.Close()
 			if mimeType == "" {
-				mimeType = detectMimeType(file)
+				mimeType = api.DetectContentType(file)
```

Comment:
i wonder if we really need to open a new file handle inside of `api.DetectContentType`. at this point we already have `f` as an open file handle. maybe we could use that instead. WDYT?

Reply:
I found problem: 
- If DetectContentType func will accept *File, then it must current preserve position in file
- Here we working with `*swarm.File` not `*os.File` which has no Seek() method to save and restore position

It doesn't look possible to get content probe from *swarm.File 
Maybe you have an idea?

Reply:
Ah, *swarm.File it's local file, then i can cast it to Seeker

Reply:
you can also try to change the `swarm.File` spec to be
```
Reader,
Closer,
Seeker
```
instead of just `ReadCloser`. that might make your life easier. but you also might need to move the type around a bit because this will probably generate a circular dependency when you'll try to reference `client` from `api` :)